lib/phoenix/endpoint.ex

```diff
@@ -814,6 +814,10 @@ defmodule Phoenix.Endpoint do
           config in runtime `{MyAppWeb.Auth, :get_session_config, []}`. Otherwise
           the session will be `nil`.
 
+        * `{:csrf_token_session_key, csrf_token_session_key}` - Name of the session
```

Comment:
I would move this inside the `session_config`, as this is session specific. :) And we can call it `csrf_token_key`.

Reply:
I'm having some trouble with this. The original `session_config` is lost because `Transport.load_config` calls `init_session/1` which in turn calls `Plug.Session.Store.init/1` which is not guaranteed to pass through any options it doesn't care about.

I thought about putting `:csrf_token_key` back in, but `Plug.Session.Store.init/1` doesn't even have to return a keyword list or map.

I ended up wrapping the return of `init_session/1`. I think it's kind of hacky though, what do you think?